x-pack/legacy/plugins/apm/server/routes/create_apm_api.ts

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { indexPatternRoute } from './index_pattern';
+import {
+  errorDistributionRoute,
+  errorGroupsRoute,
+  errorsRoute
+} from './errors';
+import {
+  serviceAgentNameRoute,
+  serviceTransactionTypesRoute,
+  servicesRoute
+} from './services';
+import {
+  agentConfigurationRoute,
+  agentConfigurationSearchRoute,
+  createAgentConfigurationRoute,
+  deleteAgentConfigurationRoute,
+  legacyAgentConfigurationSearchRoute,
+  listAgentConfigurationEnvironmentsRoute,
+  listAgentConfigurationServicesRoute,
+  updateAgentConfigurationRoute
+} from './settings';
+import { metricsChartsRoute } from './metrics';
+import { tracesRoute, tracesByIdRoute } from './traces';
+import {
+  transactionGroupsBreakdownRoute,
+  transactionGroupsChartsRoute,
+  transactionGroupsDistributionRoute,
+  transactionGroupsRoute
+} from './transaction_groups';
+import {
+  errorGroupsLocalFiltersRoute,
+  metricsLocalFiltersRoute,
+  servicesLocalFiltersRoute,
+  tracesLocalFiltersRoute,
+  transactionGroupsLocalFiltersRoute,
+  transactionsLocalFiltersRoute,
+  uiFiltersEnvironmentsRoute
+} from './ui_filters';
+import { createApi } from './create_api';
+
+const createApmApi = () => {
+  const api = createApi()
+    .add(indexPatternRoute)
+    .add(errorDistributionRoute)
+    .add(errorGroupsRoute)
+    .add(errorsRoute)
+    .add(metricsChartsRoute)
+    .add(serviceAgentNameRoute)
+    .add(serviceTransactionTypesRoute)
+    .add(servicesRoute)
+    .add(agentConfigurationRoute)
+    .add(agentConfigurationSearchRoute)
+    .add(createAgentConfigurationRoute)
+    .add(deleteAgentConfigurationRoute)
+    .add(legacyAgentConfigurationSearchRoute)
+    .add(listAgentConfigurationEnvironmentsRoute)
+    .add(listAgentConfigurationServicesRoute)
+    .add(updateAgentConfigurationRoute)
+    .add(tracesRoute)
+    .add(tracesByIdRoute)
+    .add(transactionGroupsBreakdownRoute)
+    .add(transactionGroupsChartsRoute)
+    .add(transactionGroupsDistributionRoute)
+    .add(transactionGroupsRoute)
+    .add(errorGroupsLocalFiltersRoute)
+    .add(metricsLocalFiltersRoute)
+    .add(servicesLocalFiltersRoute)
+    .add(tracesLocalFiltersRoute)
+    .add(transactionGroupsLocalFiltersRoute)
+    .add(transactionsLocalFiltersRoute)
+    .add(uiFiltersEnvironmentsRoute);
```

Comment:
Is there any order to the routes? Perhaps we can group logically or alphabetically. If you group them logically it might make sense to start each logical group with a comment.

Reply:
Not really, but what I was thinking, maybe there's an approach where we can have each file register its own routes into an `api` vessel, and then merge the `api`s in this file. Something like:

```ts
createApi()
	.merge(tracesApi)
	.merge(settingsApi)
	.merge(transactionGroupsApi)
```

Reply:
I don't mind them all in one place if they were just grouped.